### PR TITLE
Feature/ch17081/apigateway

### PR DESCRIPTION
**Pull request by richardbahari**
This is related to ticket ch17081, I am making some changes to the Java SDK to allow the usage of OAuth2 tokens. 
The assumption is that there will be paths in the gateway marked with v2 which will start using OAuth2.

If the path is v1, the current flow of using non OAuth2 is still good to use for now.

Once you change to the v2, you will be required to supply the given OAuth2 values given to you after you register.
The context.properties should have the placeholders marked where you should be putting those values.

There is a sample for OAuth2 to refresh the access token, run this after you have put the values in the context.properties, the example shall make a call to the oauth2 provider and refreshes your access token given an existing refresh token.

### CODE REVIEW

File: overledger-sdk-api/src/main/java/network/quant/util/Oauth2RefreshResponse.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    @Override
+    public String toString() {
+        return "Oauth2RefreshResponse{" +
+                "id_token='" + id_token + '\'' +
+                ", access_token='" + access_token + '\'' +
+                ", token_type='" + token_type + '\'' +
+                ", expires_in='" + expires_in + '\'' +
+                '}';
+    }
+}
```
Comment (alexchiriac) [inline overledger-sdk-api/src/main/java/network/quant/util/Oauth2RefreshResponse.java:65]: Missing newline

File: overledger-sdk-api/src/main/resources/context.properties [hunk 2]
```diff
@@ -15,9 +15,12 @@ overledger.subscribe.event = /webhook/subscribe-event
...
+overledger.mappid=network.quant.testnet
 bitcoin.faucet.url=/faucet/fund/bitcoin/{address}/{amount}
 ethereum.faucet.url=/faucet/fund/ethereum/{address}/{amount}
 ripple.faucet.url=/faucet/fund/ripple/{address}/{amount}
-overledger.fee.estimation=/fee/{dltName}/{blockNumber}
\ No newline at end of file
+overledger.fee.estimation=/fee/{dltName}/{blockNumber}
+oauth2.access.token=abc
+oauth2.refresh.token=xyz
+oauth2.id.token=jim1
```
Comment (alexchiriac) [inline overledger-sdk-api/src/main/resources/context.properties:26]: Missing newline

**Review by alexchiriac — COMMENTED** [review]
Looking good! But since you have the SDK fresh in your mind now, could you confirm if we can delete that src/main folder from the root of the project? Or is it still needed besides all the other packages?

**Review by theoQuant — APPROVED** [review]
Looks good!

**Review by kashif-quant — APPROVED** [review]

**Review by mahsa-bytt — APPROVED** [review]

File: src/main/resources/context.properties [hunk 3]
```diff
@@ -1,4 +1,6 @@
-overledger.testurl=https://bpi.testnet.overledger.io/v1
+#please replace the testurl or mainurl to point to the API Gateway with /v1 or /v2 or whichever version. Post v1 we use OAuth2
+#overledger.testurl=https://<address>/<version>
+overledger.testurl=http://api.devnet.overledger.io/v1
```
Comment (alexchiriac) [inline src/main/resources/context.properties:3]: Shouldn't the testurl be bpi.testnet, instead of api.devnet?

**Comment by richardbahari** [comment]
cancelled

**Pull request closed without merging**